solr/core/src/java/org/apache/solr/cloud/SizeLimitedDistributedMap.java

```diff
@@ -72,7 +72,7 @@ protected boolean lessThan(Long a, Long b) {
 
       for (String child : children) {
         Stat stat = zookeeper.exists(dir + "/" + child, null, true);
-        if (stat.getMzxid() <= topElementMzxId) {
+        if (stat != null && stat.getMzxid() <= topElementMzxId) {
```

Comment:
Looks like it happened because of FS's uberseer message eating

Reply:
Interesting. So it's possible to induce an overseer hang. Good catch!